cub/cub/device/dispatch/dispatch_select_if.cuh

```diff
@@ -402,21 +573,49 @@ struct DispatchSelectIf : SelectedPolicy
 
     using VsmemHelperT = cub::detail::vsmem_helper_default_fallback_policy_t<
       Policy,
-      detail::agent_select_if_wrapper_t<KEEP_REJECTS, MayAlias>::template agent_t,
+      detail::select::agent_select_if_wrapper_t<KEEP_REJECTS, MayAlias>::template agent_t,
       InputIteratorT,
       FlagsInputIteratorT,
       SelectedOutputIteratorT,
       SelectOpT,
       EqualityOpT,
-      OffsetT>;
+      per_partition_offset_t,
+      streaming_context_t>;
+
+    // Return for empty problem (also needed to avoid division by zero)
+    // TODO(elstehle): In this case d_num_selected_out will never be written. Maybe we want to write it despite?
```

Comment:
Yes, agreed, that's important. I restored the original behavior for empty problems. I also changed the tests so that is caught more easily. Would be great if you could give it another look.